Projects/Server/Network/NetState/NetState.cs

```diff
@@ -618,26 +618,46 @@ public void HandleReceive()
                                 else if (length >= 4)
                                 {
                                     int seed = (packetId << 24) | (packetReader.ReadByte() << 16) | (packetReader.ReadByte() << 8) | packetReader.ReadByte();
-
                                     if (seed == 0)
                                     {
                                         HandleError(0, 0);
                                         return;
                                     }
-
                                     _seed = seed;
                                     packetLength = 4;
-
                                     _parserState = ParserState.AwaitingNextPacket;
-                                    _protocolState = ProtocolState.GameServer_AwaitingGameServerLogin;
+                                    if (packetId == 0xFF)
+                                    {
+                                        _parserState = ParserState.ProcessingPacket;
+                                        _parserState = HandlePacket(packetReader, packetId, out packetLength);
+                                        if (_parserState == ParserState.AwaitingNextPacket)
+                                        {
+                                            _protocolState = ProtocolState.LoginServer_AwaitingKrAck;
+                                        }
+                                    }
+                                    else
+                                    {
+                                        _protocolState = ProtocolState.GameServer_AwaitingGameServerLogin;
+                                    }
                                 }
                                 else
                                 {
                                     _parserState = ParserState.AwaitingPartialPacket;
                                 }
-                                break;
+                            break;
                             }
 
+                        case ProtocolState.LoginServer_AwaitingKrAck:
+                            {
+                                if (packetId == 0xE4)
+                                {
+                                    Console.WriteLine("<---- Kr Ack Packet Received 0xE4");
```

Comment:
Don't forget to remove this before merging.

Reply:
Thanks, removed it.